third_party/forked/golang/reflect/deep_equal.go

```diff
@@ -112,10 +114,17 @@ func (e Equalities) deepValueEqual(v1, v2 reflect.Value, visited map[visit]bool,
 	if fv, ok := e[v1.Type()]; ok {
 		return fv.Call([]reflect.Value{v1, v2})[0].Bool()
 	}
-	if v1.CanAddr() {
-		if fv, ok := e[v1.Addr().Type()]; ok {
```

Comment:
don't we need to preserve the old path for `e[v1.Addr().Type()]` as well?

Reply:
needs tests exercising the existing path (to make sure it keeps working), and the new path

Reply:
added tests. These removed lines here were added during the `k8s.io/utils` move accidently. There is no old path now to fix.